client/my-sites/plugins/plugin-list-header/index.jsx

```diff
@@ -71,16 +73,20 @@ export default React.createClass( {
 	},
 
 	unselectOrSelectAll() {
+		const sites = this.props.isWpCom
+			? this.props.sites.getSelectedOrWPComBusiness()
+			: this.props.sites.getSelectedOrAllJetpackCanManage();
```

Comment:
See my note above about lazy/eager evaluation. Both of these are run _before_ checking if `this.props.isWpCom` is truthy